src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1970,6 +1970,28 @@ internal MethodSymbol EnsureInlineArrayAsSpanExists(SyntaxNode syntaxNode, Named
                                                       diagnostics);
         }
 
+        internal NamedTypeSymbol EnsureInlineArrayTypeExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, int arrayLength, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(Compilation.Assembly.RuntimeSupportsInlineArrayTypes);
+
+            string typeName = PrivateImplementationDetails.GetInlineArrayTypeName(arrayLength);
+            var privateImplClass = GetPrivateImplClass(syntaxNode, diagnostics);
+            var typeAdapter = privateImplClass.GetType(typeName);
+
+            if (typeAdapter is null)
```

Comment:
Do we have a test with two collection expressions of the same size? (the InlineArray type will be re-used/shared)

Reply:
`CollectionBuilder_InlineArrayTypes`